client.go

```diff
@@ -209,11 +219,64 @@ func (c *Client) registerProviders() {
 	driverGoFishDell := dell.New(c.Auth.Host, c.Auth.User, c.Auth.Pass, c.Logger, dellGofishOpts...)
 	c.Registry.Register(dell.ProviderName, redfish.ProviderProtocol, dell.Features, nil, driverGoFishDell)
 
-	// register supermicro vendorapi provider
+	return nil
+}
+
+// register supermicro vendorapi provider
+func (c *Client) registerSupermicroProvider() error {
```

Comment:
this func will never return an error. i would suggest not having any return value.